lib/fabrics/ofi/src/internal/RDMInitiator.hpp

```diff
@@ -0,0 +1,192 @@
+         */
+        RDMInitiatorTarget(std::unique_ptr<EgressProtocol> proto, TargetInfo remoteInfo);
+
+        /** \brief Returns true if the endpoint is idle and could be actived.
+         */
+        [[nodiscard]]
+        bool isIdle() const noexcept;
+
+        /** \brief Try to activate the endpoint.
+         *
+         * The endpoint is activated by adding the remote address to the address vector.
+         */
+        void activate(Endpoint& ep);
+
+        /** \brief Initiate a shutdown process.
+         *
+         * The endpoint will have pending work until a shutdown or error event will be received. After which it can be evicted from the initiator.
+         */
+        void shutdown(Endpoint& ep);
+
+        /** \brief Post a data transfer request to this endpoint.
+         */
+        void transfer(Endpoint& ep, std::uint64_t localIndex, std::uint64_t remoteIndex, std::uint64_t remotePayloadOffset,
+            SliceRange const& sliceRange);
+
+        /** \brief Returns true if there's pending work for this target.
+         */
+        [[nodiscard]]
+        bool hasPendingWork() const noexcept;
+
+        /** \brief Handle a completion event for this endpoint.
+         */
+        void handleCompletion(Endpoint& ep, Completion completion);
+
+    private:
+        /** \brief The idle state.
+         *
+         * In this state the endpoint waits to be activated.
+         */
+        struct Idle
+        {};
+
+        /** \brief The endpoint activated state.
+         *
+         * In this state, the remote endpoint was added to the address vector, meaning we can write to the remote endpoint.
+         */
+        struct Activated
+        {
+            ::fi_addr_t fiAddr; /**< Address index in address vector. */
+        };
+
+        /** \brief The endpoint is done and can be evicted from the initiator.
+         */
+        struct Done
+        {};
+
+        /** \brief The various states that the endpoint can be in are stored inside a variant that we move from and then back into when processing
+         * events.
+         */
+        using State = std::variant<Idle, Activated, Done>;
+
+    private:
+        State _state; /**< The current state of the endpoint. */
+        std::unique_ptr<EgressProtocol> _proto;
+
+        TargetInfo _remoteInfo;
+    };
+
+    /** \brief An initiator that uses reliable datagram (RDM) endpoints for data transfers.
+     *
+     * RDM endpoints provide connectionless communication with reliability guarantees. An address vector is used to manage destination addresses.
+     */
+    class RDMInitiator : public Initiator
+    {
+    public:
+        static std::unique_ptr<RDMInitiator> setup(mxlFabricsInitiatorConfig const&);
+
+        /** \copydoc Initiator::addTarget()
+         */
+        void addTarget(TargetInfo const&) override;
+
+        /** \copydoc Initiator::removeTarget()
+         */
+        void removeTarget(TargetInfo const&) override;
```

Comment:
`virtual` + `override`.
Method/function declarations in headers should name their parameters.